src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedCatchParameterShouldBeUnnamedCheck.java

```diff
@@ -0,0 +1,307 @@
+                        || shouldCheckDotChild(identAST))) {
+            checkIdentifier(identAST);
+        }
+    }
+
+    /**
+     * Check if the given {@link TokenTypes#IDENT} is a child of a dot operator
+     * and should be checked for usage.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     * @return true if the given {@link TokenTypes#IDENT} is a child of a dot operator
+     *     and should be checked for usage.
+     */
+    private static boolean shouldCheckDotChild(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return parent.getType() == TokenTypes.DOT
+                && identAst == parent.getFirstChild();
+    }
+
+    /**
+     * Check the given {@link TokenTypes#IDENT} and
+     * register it as used if it is the catch parameter.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     */
+    private void checkIdentifier(DetailAST identAst) {
+        if (catchParameter != null
+                && !isLeftHandSideValue(identAst)
+                && identAst.getText().equals(catchParameter.getName())) {
+            catchParameter.registerAsUsed();
+        }
+    }
+
+    /**
+     * Check if the given {@link TokenTypes#IDENT} is a left hand side value.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     * @return true if the given {@link TokenTypes#IDENT} is a left hand side value.
+     */
+    private static boolean isLeftHandSideValue(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return parent.getType() == TokenTypes.ASSIGN
+                && identAst != parent.getLastChild();
+    }
+
+    /**
+     * Log the catch parameter if it is not used after
+     * leaving its scope.
+     *
+     * @param scopeAst ast of the current scope
+     */
+    private void logUnusedCatchParameter(DetailAST scopeAst) {
+        if (catchParameter != null
+                && scopeAst == catchParameter.getScope()
+                && !catchParameter.isUsed()) {
+
+            log(catchParameter.getParameterDefAst(),
+                    MSG_UNUSED_CATCH_PARAMETER,
+                    catchParameter.getName());
+        }
+    }
+
+    /**
+     * Maintains information about the catch parameter.
+     */
+    private static final class CatchParameter {
+
+        /**
+         * The name of the catch parameter.
+         */
+        private final String name;
+
+        /**
+         * The scope of the catch parameter is determined by the ast of type
+         * {@link TokenTypes#LITERAL_CATCH}.
+         */
+        private final DetailAST scope;
+
+        /**
+         * Ast of type {@link TokenTypes#PARAMETER_DEF} to use it when logging.
+         */
+        private final DetailAST parameterDefAst;
+
+        /**
+         * Is the variable used.
+         */
+        private boolean used;
```

Comment:
```suggestion
        private boolean isUsed;
```

Reply:
done

Reply:
`[WARNING] PMD Failure: com.puppycrawl.tools.checkstyle.checks.coding.CatchParameterDetails:211 Rule:AvoidFieldNameMatchingMethodName Priority:3 Field isUsed has the same name as a method.`

reverted to used. Do you agree? I am not with changing the method name from `isUsed()`

Reply:
I am good with reverting it, it's not a big deal